docker-compose.solr.yaml

```
@@ -2,7 +2,10 @@
 
 services:
   solr:
-    build: ./solr
+    build: 
+      context: ./solr
+      args:
+        SOLR_BASE_IMAGE: ${SOLR_BASE_IMAGE}
```

> I see the tests failing with warnings that $SOLR_BASE_IMAGE should not be blank.

```suggestion
        SOLR_BASE_IMAGE: ${SOLR_BASE_IMAGE:-solr:9.6}
```

---

If we do this then the default assignment in the Dockerfile should be removed, since this one will do the job.

Another approach here would be this maybe:

```suggestion
        SOLR_BASE_IMAGE: "${SOLR_BASE_IMAGE}"
```

---

> If we do this then the default assignment in the Dockerfile should be removed

No, it should be in the `Dockerfile` to prevent this warning https://docs.docker.com/reference/build-checks/invalid-default-arg-in-from/

And we need it in `docker-compose.solr.yaml` to easily identify what is the default without looking into other files.

The same approach is used in https://github.com/ddev/ddev-opensearch

---

The problem is that if it's in both places, they have to be separately maintained and it's not likely to be done. Maybe change the default in the Dockerfile to `scratch` to make that problem not happen?